src/providers/dataProvider/index.ts

```diff
@@ -36,10 +48,27 @@ export const generateBatchId = async (
     pagination: { page: 1, perPage: 1000 },
     filter: { yearOfReceipt: year }
   })
-  return batches.data.length.toLocaleString('en-US', {
-    minimumIntegerDigits: 2,
-    useGrouping: false
-  })
+
```

Comment:
hmm, for the next line - I understand why we have to handle a zero length array (since `reduce` will fail).  But, we don't have to handle `1` here, do we?  I _guess_ we can just let the `reduce` method run on the single item.

The pros and cons are probably 50/50 on it, but I _think_ I favour just using this `special handling` for an empty array.  Aah, plus...  if we're just handing a zero length array, we can return `00`, without considering the array length.